### DISCUSSION TITLE
Why is nil session handled only partly in retryable/write_worker which was added from 2.19.0?

### DISCUSSION THREAD

**Discussion by raulalexe** [mongodb/mongo-ruby-driver#2811] [Q&A]
I came across an issue while using version 2.19.2 and 2.19.3 with mongodb 3.6 in a setup with a primary and 2 replicas.
In write_worker.rb line 107 there is this bit of code `session&.client.options[:retry_writes]` which seems to have been thought to handle a `nil` session but not implemented completely as when the session is nil the `client` will be nil and will crash on accessing `options`. If I downgrade to 2.18.3 it's working fine as this bit of code isn't there.

**Comment by comandeo** [comment]
Hi @raulalexe , thank you for reporting this! I created JIRA ticker for the problem - https://jira.mongodb.org/browse/RUBY-3358 - to have the team review this and identify next steps.

**Reply by cynthia-anya** [reply]
Hi @comandeo, this issue is preventing us from upgrading. Is there an ETA on a fix for this?

**Reply by comandeo** [reply]
Hi @cynthia-anya , we increase the priority of this. We should be able to ship a fixed version within a couple of weeks.
